src/sds_data_model/dataframe.py

```diff
@@ -249,3 +253,46 @@ def to_zarr(
             .mode("overwrite")
             .save()
         )
+
+    def index(
+        self: Self,
+        resolution: int,
+        how: str = "intersects",
+        index_column_name: str = "bng",
+        bounds_column_name: str = "bounds",
+        geometry_column_name: str = "geometry",
+        exploded: bool = True,
+    ) -> _DataFrameWrapper:
```

Comment:
I thought we'd be able to use the `Self` type here that was first proposed in [PEP 673](https://peps.python.org/pep-0673/), is currently available through [`typing-extensions`](https://pypi.org/project/typing-extensions/) and will, apparently, become part of the standard library in Python 3.11, but `mypy` doesn't like it, so we have to use `_DataFrameWrapper` for now.

```suggestion
    def index(
        self: _DataFrameWrapper,
        resolution: int,
        how: str = "intersects",
        index_column_name: str = "bng",
        bounds_column_name: str = "bounds",
        geometry_column_name: str = "geometry",
        exploded: bool = True,
    ) -> _DataFrameWrapper:
```